lang/java/reef-webserver/src/main/java/org/apache/reef/webserver/HttpServerImpl.java

```diff
@@ -109,15 +116,16 @@
   private Server tryPort(final int portNumber) throws Exception {
     Server srv = new Server();
     final Connector connector = new SocketConnector();
-    connector.setHost(addressProvider.getLocalAddress());
+    connector.setHost(this.hostAddress);
     connector.setPort(portNumber);
     srv.addConnector(connector);
     try {
       srv.start();
       LOG.log(Level.INFO, "Jetty Server started with port: {0}", portNumber);
     } catch (final BindException ex) {
       srv = null;
-      LOG.log(Level.FINEST, "Cannot use port: {0}. Will try another", portNumber);
+      LOG.log(Level.FINEST,
+          String.format("Cannot use host:%s,port: %s. Will try another", this.hostAddress, portNumber));
```

Comment:
use `{0}` interpolation in logging instead of `String.format()`:
```java
LOG.log(Level.FINEST, "Cannot use host: {0}, port: {1}. Will try another.",
    new Object[] {this.hostAddress, portNumber});
```

Reply:
Fixed.

---
In reply to: [207103982](https://github.com/apache/reef/pull/1478#discussion_r207103982) [](ancestors = 207103982)